sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -339,8 +339,11 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with Logging {
       case SqlBaseParser.SELECT =>
         // Regular select
 
+        // Add hints.
+        val withHint = relation.optionalMap(ctx.hint)(withHints)
```

Comment:
according to the grammar: `SELECT /*+ hint... */ ...`, it seems that hint should be applied to the whole SELECT clause, not only its relations?